artemis-log-annotation-processor/src/main/java/org/apache/activemq/artemis/logs/annotation/processor/LogAnnotationProcessor.java

```diff
@@ -204,6 +208,17 @@ void validateRegexID(LogBundle bundleAnnotation, long id) {
       }
    }
 
+   void checkRetiredIDs(LogBundle bundleAnnotation, long id) {
+      if (bundleAnnotation.retiredIDs() != null && !bundleAnnotation.retiredIDs().isEmpty()) {
+         if (Arrays.stream(bundleAnnotation.retiredIDs().split(","))
+                   .map(String::trim)
+                   .map(Long::valueOf)
+                   .collect(Collectors.toList()).contains(id)) {
+            throw new IllegalArgumentException("Code ID " + id + " has been retired. Select a different ID.");
+         }
+      }
+   }
```

Comment:
Although not performance-critical, repeatedly splitting the retiredIDs String, trimming them, creating the Long's, and Collecting them to the same ultimate List over-and-over again for every LogMessage/Message annotation seems especially inefficient. It could be done once upfront for each LogBundle and the result reused?

There are a lot of sequential retired IDs. Someone hitting one seems likely to just try the next one, and find the same error, before then having to look at the bundle definition and consider the list and find the next non-retired and not-in-use one.

This bit of code has the retired IDs, meaning it can easily indicate what they are so anyone hitting the failure can immediately see the retired ones they cant use. (It could even indicate the next ID that isnt retired and could also further look at whether that is already used and suggest the next that isnt...but that would be a big change, so just printing the retired ones would be an easy help instead hehe).

Reply:
Updated. Let me know what you think.

Reply:
The 'dont create repeatedly' aspect looks better, however it still doesn't do anything to indicate about the IDs it knows you cant use, which would prevent the very likely 'just try using next ID...find it is also retired' gotcha.

Reply:
I've updated the code to suggest a valid ID when an invalid one is found.